### ISSUE TITLE
Error happened When using vs2015 compiler

### ISSUE THREAD

**Issue by FinalAgito** [stellar/stellar-core#1312]
***Error:***
 * #error asio.h is somewhat particular about when it gets included -- it wants to be the first to include <windows.h>. Include it before everything else.

***Q:***
* This problem happened when I using vs compiler to compile this program. I have tried to change _Win32  to _Win64, or  add Winsock2.h, but it's not work.
* I also tried to find Answer in **Issues** area, and check the environment of **flex** and **bison**, or **gcc**. Used **Mingw** of 64 bit version. Sadly, still not work.
* I think that maybe the code in this program have conflict with which in PC system's .

* Have anyone knows how to fix this problem?

**Comment by MonsieurNicolas** [comment]
yes, the Windows build is currently busted on master (and probably some number of releases).
The fix is in the PR queue https://github.com/stellar/stellar-core/pull/1306 . Hopefully this will get merged soon.

It will require upgrading to Visual Studio 2017 as 2015 is no longer downloadable from the Microsoft site.

**Comment by FinalAgito** [comment]
@MonsieurNicolas  
Thanks for your Answer!
So,  It means that ,vs 2015 can't  be used in this Version? I mean, If I followed steps in #1306.

**Comment by MonsieurNicolas** [comment]
The steps in #1306 are actually pretty hard to follow until that pull request gets merged as it depends on changes in multiple git repos, so I recommend just waiting.
After that, opening the project files may work with 2015, but as 2017 Community Edition is a free download, it's probably not worth the effort.
A new alternative to get going on Windows is outlined in #1306 as well, which is to use the [Windows Subsystem for Linux](https://github.com/MonsieurNicolas/stellar-core/blob/VS2017/INSTALL-Windows.md#build-on-windows-using-the-windows-subsystem-for-linux) ; performance wise it's comparable to Windows when pointing to a native Windows postgres instance. Cygwin is also an option but probably less reliable from a build point of view.

**Comment by FinalAgito** [comment]
Thanks a lot!

**Comment by MonsieurNicolas** [comment]
btw, current master should build again using VS2017
